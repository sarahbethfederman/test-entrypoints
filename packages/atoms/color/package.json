{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  