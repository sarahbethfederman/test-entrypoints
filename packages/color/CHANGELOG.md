2.0.2 - New Changelog

## [3.0.0] - 2019-04-30 - [HUB-54] (https://creditandfinance.atlassian.net/browse/HUB-54)
 
### Deleted
- Remove the CJS bundles.