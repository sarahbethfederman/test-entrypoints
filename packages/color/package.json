{
  "name": "@lendi-ui/color",
  "version": "3.0.0",
  "description": "Color utilities for color.",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^6.0.0",
    "styled-components": "^3"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^2.0.0"
  },
  "scripts": {
    "build:esm": "tsc --project tsconfig.esm.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:esm"
  }
}
