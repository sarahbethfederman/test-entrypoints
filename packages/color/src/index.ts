import { css } from 'styled-components';
import { select } from '@lendi-ui/theme';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';

export type NameOrNameMap = BreakpointValue<string> | BreakpointValueMap<string>;

export function color(name: NameOrNameMap) {
  return select(`colors.${name}`);
}

export function fg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            color: ${color(n)};
          `
  );
}

export function bg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            background-color: ${color(n)};
          `
  );
}
