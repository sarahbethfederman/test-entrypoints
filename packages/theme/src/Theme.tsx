import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
