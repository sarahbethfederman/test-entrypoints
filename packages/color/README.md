# @lendi-ui/color

Color style with `bg`(background color), `fg`(font color) and `color`(every CSS property relate to color).

## Installation

```
yarn add @lendi-ui/color
```

## Usage

- `bg` for styling background-color.

```jsx
import { bg } from '@lendi-ui/color';
const Square = styled.div`
  width: 2em;
  height: 2em;
  ${bg('warn.500')};
`;
```

- `fg` for styling font-color.

```jsx
import { fg } from '@lendi-ui/color';
const WarningText = styled.span`
  ${fg('warn.500')};
`;
```

- `color` for styling every CSS porperty relate to color.

```jsx
import { color } from '@lendi-ui/color';
const Circle = styled.div`
  width: 0;
  border-radius: 1em;
  border: 1em solid ${color('primary.500')};
`;
```
