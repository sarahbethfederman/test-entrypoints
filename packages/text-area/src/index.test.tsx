import * as React from 'react';
import { mount } from 'enzyme';
import Theme from '@lendi-ui/theme';
import TextArea from './index';
import { Size, TextAreaWrapper } from './index.style';
import { color } from '@lendi-ui/color';

const sizes: Size[] = ['lg', 'md', 'sm'];

let element;

function render(props) {
  element = mount(
    <Theme>
      <TextArea value={props.value} placeholder="input here ..." onChange={jest.fn()} {...props} />
    </Theme>
  );
}

describe('Text area', () => {
  it('should render the text area component', () => {
    render({});
    expect(element.find(TextArea)).toHaveLength(1);
    expect(element.find(TextArea)).toMatchSnapshot();
    expect(element.find(TextAreaWrapper)).toHaveLength(1);
  });

  describe('variations of the text area in all given sizes', () => {
    sizes.forEach((size) => {
      describe(`${size}`, () => {
        it('should render 100% width of with isFullWidth prop', () => {
          const isFullWidth = true;
          render({ isFullWidth });
          expect(element.find(TextAreaWrapper)).toHaveStyleRule('width', '100%');
          expect(element.find(TextAreaWrapper)).toMatchSnapshot();
        });

        it('should render the component in a disabled state with the isDisabled prop', () => {
          const isDisabled = true;
          render({ isDisabled });
          expect(element.find(TextAreaWrapper)).toHaveStyleRule(`'background-color', ${color('shade.25')}`);
          expect(element.find(TextAreaWrapper)).toMatchSnapshot();
        });

        it('should render the component with a white border and transparent background if the isInverse prop is true', () => {
          const isInverse = true;
          render({ isInverse });
          expect(element.find(TextAreaWrapper)).toHaveStyleRule('background-color', 'transparent');
          expect(element.find(TextAreaWrapper)).toHaveStyleRule(`'border', '1px solid ${color('shade.25')}'`);
          expect(element.find(TextAreaWrapper)).toMatchSnapshot();
        });

        it('should render the component with a red border if the isError state is true', () => {
          const isError = true;
          render({ isError });
          expect(element.find(TextAreaWrapper)).toHaveStyleRule(`'border', '1px solid ${color('warn.500')}`);
          expect(element.find(TextAreaWrapper)).toMatchSnapshot();
        });
      });
    });
  });

  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({
        'aria-label': ARIA_LABEL,
        'aria-describedby': ARIA_DESCRIBE_BY,
      });
      const attributes = element.find('textarea').props();
      expect(attributes['aria-label']).toBe(ARIA_LABEL);
      expect(attributes['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id, tabIndex', () => {
      const TEXT_ID = 'testId';
      render({
        id: TEXT_ID,
        tabIndex: 1,
      });
      const attributes = element.find('textarea').props();
      expect(attributes.id).toBe(TEXT_ID);
      expect(attributes.tabIndex).toBe(1);
    });
  });
});
