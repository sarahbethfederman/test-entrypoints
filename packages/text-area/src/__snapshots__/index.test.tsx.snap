// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text area should render the text area component 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #c8c8c8;
  background-color: #ffffff;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #929292;
}

<TextArea
  onChange={[MockFunction]}
  placeholder="input here ..."
>
  <styled.textarea
    className=""
    disabled={false}
    isDisabled={false}
    isError={false}
    isFullWidth={false}
    isInverse={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  >
    <textarea
      className="c0"
      disabled={false}
      onChange={[MockFunction]}
      placeholder="input here ..."
      size="md"
    />
  </styled.textarea>
</TextArea>
`;

exports[`Text area variations of the text area in all given sizes lg should render 100% width of with isFullWidth prop 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: 100%;
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #c8c8c8;
  background-color: #ffffff;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #929292;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={false}
  isFullWidth={true}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes lg should render the component in a disabled state with the isDisabled prop 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  cursor: not-allowed;
  background-color: #f8f8f8;
  border: 1px solid #e3e3e3;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #929292;
}

<styled.textarea
  className=""
  disabled={true}
  isDisabled={true}
  isError={false}
  isFullWidth={false}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={true}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes lg should render the component with a red border if the isError state is true 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #ff1c4a;
  background-color: #ffffff;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #ff1c4a;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={true}
  isFullWidth={false}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes lg should render the component with a white border and transparent background if the isInverse prop is true 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #ffffff;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #f8f8f8;
  background-color: transparent;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #ffffff;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={false}
  isFullWidth={false}
  isInverse={true}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes md should render 100% width of with isFullWidth prop 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: 100%;
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #c8c8c8;
  background-color: #ffffff;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #929292;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={false}
  isFullWidth={true}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes md should render the component in a disabled state with the isDisabled prop 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  cursor: not-allowed;
  background-color: #f8f8f8;
  border: 1px solid #e3e3e3;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #929292;
}

<styled.textarea
  className=""
  disabled={true}
  isDisabled={true}
  isError={false}
  isFullWidth={false}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={true}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes md should render the component with a red border if the isError state is true 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #ff1c4a;
  background-color: #ffffff;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #ff1c4a;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={true}
  isFullWidth={false}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes md should render the component with a white border and transparent background if the isInverse prop is true 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #ffffff;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #f8f8f8;
  background-color: transparent;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #ffffff;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={false}
  isFullWidth={false}
  isInverse={true}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes sm should render 100% width of with isFullWidth prop 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: 100%;
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #c8c8c8;
  background-color: #ffffff;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #929292;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={false}
  isFullWidth={true}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes sm should render the component in a disabled state with the isDisabled prop 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  cursor: not-allowed;
  background-color: #f8f8f8;
  border: 1px solid #e3e3e3;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #929292;
}

<styled.textarea
  className=""
  disabled={true}
  isDisabled={true}
  isError={false}
  isFullWidth={false}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={true}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes sm should render the component with a red border if the isError state is true 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #474747;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #ff1c4a;
  background-color: #ffffff;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #ff1c4a;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={true}
  isFullWidth={false}
  isInverse={false}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;

exports[`Text area variations of the text area in all given sizes sm should render the component with a white border and transparent background if the isInverse prop is true 1`] = `
.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  resize: none;
  border-radius: 6px;
  box-sizing: border-box;
  font-family: 'Open Sans',sans-serif;
  color: #ffffff;
  width: calc(21.5 * var(--lendi-ui-size));
  height: calc(6 * var(--lendi-ui-size));
  padding: calc(0.75 * var(--lendi-ui-size)) var(--lendi-ui-size);
  font-size: calc(1.125 * var(--lendi-ui-size));
  border: 1px solid #f8f8f8;
  background-color: transparent;
}

.c0::-webkit-input-placeholder {
  color: #adadad;
}

.c0::-moz-placeholder {
  color: #adadad;
}

.c0:-ms-input-placeholder {
  color: #adadad;
}

.c0::placeholder {
  color: #adadad;
}

.c0:hover:not(:focus) {
  border: 1px solid #ffffff;
}

<styled.textarea
  className=""
  disabled={false}
  isDisabled={false}
  isError={false}
  isFullWidth={false}
  isInverse={true}
  onChange={[MockFunction]}
  placeholder="input here ..."
  size="md"
>
  <textarea
    className="c0"
    disabled={false}
    onChange={[MockFunction]}
    placeholder="input here ..."
    size="md"
  />
</styled.textarea>
`;
